techniques/system/common/1.0/promises.st

```diff
@@ -235,9 +238,34 @@ bundle agent check_disable_agent
       "FATAL: The file ${g.rudder_disable_agent_file} is present. Rudder will kill all running daemons and halt immediately."
         classes => if_ok("abort_report_done"),
         action  => immediate;
+}
+
 
+##########################################################
+# Check for "force-audit-agent" file and cleanly stop and
+# warn about this if it is present and agent is not in audit
+#
+# All promises in this bundle MUST have action=>immediate,
+# to avoid locking issues in case of several close runs.
+##########################################################
+bundle agent check_rudder_force_audit_agent
+{
+  classes:
+      "should_be_in_audit" expression => fileexists("${g.rudder_force_audit_agent}");
+
+      # Only define this class when we're ready to die - this is a special class name in "abortclasses"
+      "should_not_continue"
```

Comment:
could you use the new abort mechanism instead?

Reply:
Like in ```bundle agent rudder_check_agent_version   
{
  methods:  
    cfengine_3_1|cfengine_3_2|cfengine_3_3|cfengine_3_4|cfengine_3_5|cfengine_3_6|cfengine_3_7|cfengine_3_8|cfengine_3_9::
      "any" usebundle => _abort("unsupported_agent", "This agent is not compatible with its Rudder server, please upgrade");
}
```

Reply:
(it send a control report to the webapp before aborting, which may be useful in the future)

Reply:
i had no idea it even existed